### ISSUE TITLE
Unable to run EFA jobs on EKS Ubuntu nodes

### ISSUE THREAD

**Issue by suruchissingh** [aws-samples/aws-efa-eks#10]
We have an EKS cluster where we are trying to run EFA enabled jobs on ubuntu nodes, but unable to get this running due to networking issues on pods running on EFA enabled nodes. The error we see are usually related to pods communication , pods scheduled on EFA nodes are unable to communicate with each other.
We have observed that we are facing the issues only with EFA enabled ubuntu nodes , the setup is working fine for amazon linux nodes. We have been able to use EFA in non EKS EC2 ubuntu machines in the past but for some reason this is not working with EKS ubuntu nodes.
As for our use case we need to use ubuntu nodes only , We would like to understand if there are any differences in how vpc cni / EFA networking works for ubuntu and amazon linux. Also are there any similar use cases with successful implementation of EFA on Ubuntu nodes in EKS. This is a major blocker for us, any recommendations/references on this would be highly appreciated.

**Comment by satishpasumarthi** [comment]
> We have been able to use EFA in non EKS EC2 ubuntu machines in the past but for some reason this is not working with EKS ubuntu nodes.

@suruchissingh What CNI were you using for non EKS EC2 ubuntu machines?

**Comment by willgleich** [comment]
I encountered similar timeout issues when deploying efas on ubuntu. I was able to resolve the issues by deleting the default and internal vpc cidr routes on all non primary interfaces. If you compare the IP routes on AL2 vs Ubuntu you'll see that TCP traffic gets routed over all the efas on ubuntu, but only the eth0 on AL2. If you make the routes look similar it resolved the issue for me
`ip route`

```
ip route del default via $GATEWAY dev ens65
ip route del default via $GATEWAY dev ens130
ip route del default via $GATEWAY dev ens163
ip route del $CIDR dev ens65
ip route del $CIDR dev ens130
ip route del $CIDR dev ens163
```
